taverna-cwl-activity-ui/src/main/java/org/apache/taverna/cwl/ui/serviceprovider/CwlServiceDesc.java

```diff
@@ -16,79 +16,67 @@
  *******************************************************************************/
 package org.apache.taverna.cwl.ui.serviceprovider;
 
-import java.util.Arrays;
 import java.util.List;
-import java.util.Map;
 
 import javax.swing.Icon;
 
-import org.apache.taverna.cwl.CwlActivityConfigurationBean;
 import org.apache.taverna.cwl.CwlDumyActivity;
 
+import com.fasterxml.jackson.databind.JsonNode;
+
 import net.sf.taverna.t2.servicedescriptions.ServiceDescription;
 import net.sf.taverna.t2.workflowmodel.processor.activity.Activity;
 
-public class CwlServiceDesc extends ServiceDescription<CwlActivityConfigurationBean> {
+public class CwlServiceDesc extends ServiceDescription<JsonNode > {
```

Comment:
+1 for the move to `JsonNode` - except in Taverna 3 you should not need the `<JsonNode>` generics - rather the configuration is always an [`Configuration`](https://taverna.incubator.apache.org/javadoc/taverna-language/org/apache/taverna/scufl2/api/configurations/Configuration.html) object which has `getJson*()` methods.